Helper/Order.php

```diff
@@ -754,13 +756,21 @@ public function dispatchPostCheckoutEvents($order, $quote) {
         $this->_eventManager->dispatch(
             'checkout_submit_all_after', [
                 'order' => $order,
-                'quote' => $quote
+                'quote' => $immutableQuote
             ]
         );
 
+        // Because we used the true session quote for order save, the order's quote ID
+        // does not match the immutable quote that we will keep for historical purposes.
+        // Here we reassign the quote ID to the ID of the same quote copy that we have
+        // passed to the post checkout events.  We do this after the events have processed
+        // to respect any third party code that is perhaps hard-coded to expect
+        // the session quote value in this field.
+        $order->setQuoteId($immutableQuote->getId())->save();
+
```

Comment:
@danac-gs Here is the significant change #2.  
This keeps a quote reference to the order that can be restored at any point by any plugin.  If this quote is used as an **order_reference** at Bolt, it will pass because it was formerly an immutable quote id that has never been sent to Bolt as an order reference.  In the case of quote reactivation, this cloned quote would now become the parent quote and behave exactly in the same manner as if it were always a naturally created Magento session quote.